src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/creategeneralorder/CreateGeneralOrderAboutToStartHandler.java

```diff
@@ -0,0 +1,85 @@
+package uk.gov.hmcts.reform.finrem.caseorchestration.handler.creategeneralorder;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.finrem.caseorchestration.ccd.callback.CallbackType;
+import uk.gov.hmcts.reform.finrem.caseorchestration.controllers.GenericAboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.finrem.caseorchestration.handler.FinremCallbackHandler;
+import uk.gov.hmcts.reform.finrem.caseorchestration.handler.FinremCallbackRequest;
+import uk.gov.hmcts.reform.finrem.caseorchestration.mapper.FinremCaseDetailsMapper;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.EventType;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CaseType;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.FinremCaseData;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.wrapper.GeneralOrderWrapper;
+import uk.gov.hmcts.reform.finrem.caseorchestration.service.IdamService;
+
+import java.time.Clock;
+import java.time.LocalDate;
+import java.time.ZoneId;
+
+import static uk.gov.hmcts.reform.finrem.caseorchestration.ccd.callback.CallbackType.ABOUT_TO_START;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.EventType.GENERAL_ORDER;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.EventType.GENERAL_ORDER_CONSENT_IN_CONTESTED;
+
+
+@Service
+@Slf4j
+public class CreateGeneralOrderAboutToStartHandler extends FinremCallbackHandler {
+
+    private final IdamService idamService;
+    private final Clock clock;
+
+    @Autowired
+    public CreateGeneralOrderAboutToStartHandler(FinremCaseDetailsMapper finremCaseDetailsMapper,
+                                                 IdamService idamService) {
+        this(finremCaseDetailsMapper, idamService, Clock.system(ZoneId.of("Europe/London")));
+    }
+
+    public CreateGeneralOrderAboutToStartHandler(FinremCaseDetailsMapper finremCaseDetailsMapper,
+                                                 IdamService idamService, Clock clock) {
+        super(finremCaseDetailsMapper);
+        this.idamService = idamService;
+        this.clock = clock;
+    }
+
+    @Override
+    public boolean canHandle(CallbackType callbackType, CaseType caseType, EventType eventType) {
+        if (!ABOUT_TO_START.equals(callbackType)) {
+            return false;
```

Comment:
Please move them into separate handlers. One for Contested and one for Consented.